compiler/rustc_hir_analysis/src/check/check.rs

```diff
@@ -461,7 +461,15 @@ fn check_opaque_meets_bounds<'tcx>(
     }
     match origin {
         // Checked when type checking the function containing them.
-        hir::OpaqueTyOrigin::FnReturn(..) | hir::OpaqueTyOrigin::AsyncFn(..) => {}
+        hir::OpaqueTyOrigin::FnReturn(..) | hir::OpaqueTyOrigin::AsyncFn(..) => {
```

Comment:
Wondering if AsyncFn should continue to fall through?

Reply:
hmm true. But this is a revert of any stable effects of my old PR, there may be a repro with async fn

Reply:
Ah yeah, my mind jumped directly to AFITs, not even thinking about standalone functions.